client-sdk/ts-web/rt/src/callformat.ts

```diff
@@ -1,4 +1,105 @@
+import * as oasis from '@oasisprotocol/client';
+
+import * as ed from '@noble/ed25519';
+import * as types from './types';
+import * as mrae from './mrae';
+
+var deoxysii = require('deoxysii');
+
 /**
  * Call data key pair ID domain separation context base.
  */
 export const CALL_DATA_KEY_PAIR_ID_CONTEXT_BASE = 'oasis-runtime-sdk/private: tx';
+
+/**
+ *  EncodeConfig is call encoding configuration.
+ */
+export interface EncodeConfig {
+    /**
+     * publicKey is an optional runtime's call data public key to use for encrypted call formats.
+     */
+    publicKey?: types.KeyManagerSignedPublicKey;
+}
+
+export interface metaEncryptedX25519DeoxysII {
+    sk: Uint8Array;
+    pk: Uint8Array;
+}
+
+/**
+ * encodeCall encodes a call based on its configured call format.
+ * It returns the encoded call and any metadata needed to successfully decode the result.
+ */
+export async function encodeCall(
+    call: types.Call,
+    cf: types.CallFormat,
+    cfg?: EncodeConfig,
```

Comment:
```suggestion
    format: types.CallFormat,
    config?: EncodeConfig,
```
otherwise it's too many `c`s and `f`s